pkg/asset/installconfig/aws/validation.go

```diff
@@ -211,12 +212,34 @@ func validateDuplicateSubnetZones(fldPath *field.Path, subnets map[string]Subnet
 }
 
 func validateServiceEndpoints(fldPath *field.Path, region string, services []awstypes.ServiceEndpoint) error {
```

Comment:
Make the return `field.ErrorList` so that each error can be added individually rather than aggregated in this function only to be aggregated again later.
```suggestion
func validateServiceEndpoints(fldPath *field.Path, region string, services []awstypes.ServiceEndpoint) field.ErrorList {
```
This means also changing the local error about failing to find an endpoint for a service to a `field.Invalid` error in this function rather than having the caller put the error into a `field.Invalid` error.

Reply:
Great idea. I hadn't even considered changing the overall function workflow. Done.